ykman-gui/qml/ChangePin.qml

```diff
@@ -0,0 +1,143 @@
+import QtQuick 2.5
+import QtQuick.Controls 1.4
+import QtQuick.Layouts 1.1
+
+ColumnLayout {
+
+    property string codeName: qsTr('PIN')
+    //: Input field for the current value of the code (for example PIN) to be changed
+    property string currentCodeLabel: qsTr('Current %1:').arg(codeName)
+    property int maxLength: 8
+    property int minLength: 6
+
+    property string normalColor: 'black'
+    property string errorColor: 'red'
+
+    signal accepted
+    signal canceled
+    signal codeChanged(string currentCode, string newCode)
+
+    onAccepted: {
+        if (valid(currentInput.text, newInput.text, repeatInput.text)) {
+            codeChanged(currentInput.text, newInput.text)
+            reset()
+        } else {
+            state.attemptMade = true
+        }
+    }
+
+    Component.onCompleted: reset()
```

Comment:
Good idea! I think I expected that an `onCanceled` handler defined in the component would be overridden by one defined in a use of the component, but that seems to not be the case.

In practice this already happened, because all current uses of `ChangePin` pop it off a `StackView` and destroy it when canceled, but it makes sense to include the handler anyway.